pdfarranger/metadata.py

```diff
@@ -93,17 +93,28 @@ def _safeiter(elements):
 
 
 def merge_doc(metadata, input_docs):
-    """Same as merge but with pikepdf.PDF object instead of files"""
+    """Same as merge but with pikepdf.PDF object instead of files
+
+    XMP metadata take precedence over equivalent docinfo metadata,
+    metadata of later opened files are merged into these of earlier opened ones
+    """
     r = metadata.copy()
     for doc in input_docs:
         with doc.open_metadata() as meta:
-            load_from_docinfo(meta, doc)
             for k, v in _safeiter(meta.items()):
                 if not _pikepdf_meta_is_valid(v):
                     # workaround for https://github.com/pikepdf/pikepdf/issues/84
                     del meta[k]
                 elif k not in metadata:
```

Comment:
This actually still use the old behavior (later added values overwrites previous added). Insted this could also be `elif k not in r:`

I think the reason the test files work fine for the "Title" is that the value is not i XMP but only in docinfo, and docinfo overwriting is fixed but not xmp overwriting.

Reply:
Yes, you're right. Fixed and amended test pdf files to cover this case.